### PR TITLE
feat: UX adjustments - Modules

### LINKED ISSUE #2624: Redesign Kyma Modularization to Improve User Experience and Navigation
**Description**

As [Sebastian](https://github.com/kyma-project/community/blob/main/manifesto/personas.md#sebastian---tech-lead) and [Matthew](https://github.com/kyma-project/community/blob/main/manifesto/personas.md#matthew---cloud-native-software-engineer), we would like the Kyma Modularization feature to be redesigned based on the feedback we provided.

**Acceptance Criteria**

- [x] Follow Project DOD
- [x] Integrate a hero banner on the homepage.
- [x] Add a configuration navigation node at the cluster level.
- [x] Implement card-style design for module listings based on Figma prototype.
- [x] Create dedicated landing pages for each module with additional configurations.
- [x] Utilize a 3-step navigation system for module details.
- [x] Allow in-context editing in a different tab.
- [x] Match SAP's Horizon Visual Theme.
- [x] Use UI5 Web Components.
- [x] Make sure that no modules section looks like this:
image

- [x] https://github.com/kyma-project/busola/issues/3030
- [x] https://github.com/kyma-project/busola/issues/3031
- [x] https://github.com/kyma-project/busola/issues/3028
- [x] https://github.com/kyma-project/busola/issues/3022
- [x] https://github.com/kyma-project/busola/issues/2995

**Useful Links**

- [SAP Horizon Theme Documentation](https://btpx.frontify.com/)
- [UI5 Web Components](https://sap.github.io/ui5-webcomponents/)
- [Fima Prototype](https://www.figma.com/file/AkzF9I1INdwc5iUtm5fKNU/Redesign-%26-Modules?type=design&node-id=7548-81549&mode=design&t=hil2eEwlt1a4UWbv-0)

**Reasons**

- Enhancing the user experience based on actual user feedback ensures that the platform evolves in alignment with user needs.
- The modular landing pages provide a more organized and intuitive way for users to interact with different modules.
- A card-style design will improve readability and navigation, making it easier to access crucial information.
- The updated design and navigation features are particularly beneficial for our personas, Sebastian and Matthew, who have high standards for operational support, modular architecture, and tool flexibility.

**Attachments**

image
image

image
image

### PR DESCRIPTION
**Pull request by mrCherry97**
**Description**

Changes proposed in this pull request:

- fix: title cut off
- feat: change modify to add
- feat: adjust labels for badges - remove uppercase
- feat: remove delete button
- feat: add on row click for modules
- feat: add trash icon for uninstalling the module
- feat: add max-width for all of the popovers
- feat: add hideFormHeader prop
- fix: window title for extensibility 
- feat: isSelected for modules list

**Related issue(s)**
Closes #2624

### CODE REVIEW

**Review by OliwiaGowor — COMMENTED** [review]
- [ ] some of the status badges are now all lowercase (e.g. Modules -> nats -> Conditions)
Screenshot 2024-06-12 at 14 00 18

- [ ] selection in list disappears after refresh
Screenshot 2024-06-12 at 14 04 58

- [ ] top corners on the background are not rounded (ideally remove the background)
Screenshot 2024-06-12 at 14 14 25

- [x] change `Save` to `Add` in the form
Screenshot 2024-06-12 at 14 22 06

- [ ] not all popovers have the max width (e.g. Modules -> api-gateway -> ready)
Screenshot 2024-06-12 at 14 31 51

- [ ] link text and icon are slightly misaligned on the list view
Screenshot 2024-06-12 at 14 44 52

- [ ] after going into Edit Mode on list view and going back into View tab without editing the resource you can't click anything connected to navigation (sidebar, resources on the list, etc.)

File: src/shared/components/GenericList/GenericList.js [hunk 1]
```diff
@@ -270,66 +283,71 @@ export const GenericList = ({
...
-    const selectedEntry = entries.find(entry => {
-      return (
-        entry?.metadata?.name === e.target.children[nameColIndex].innerText ||
-        pluralize(entry?.spec?.names?.kind ?? '') ===
-          e.target.children[nameColIndex].innerText ||
-        entry?.name === e.target.children[nameColIndex].innerText
-      );
-    });
+    if (customRowClick) {
+      setEntrySelected(e.target.children[0].innerText);
```
Comment (OliwiaGowor) [inline src/shared/components/GenericList/GenericList.js:287]: shouldn't it be also `e.target.children[nameColIndex].innerText`?

**Comment by OliwiaGowor** [comment]
LGTM

**Review by chriskari — COMMENTED** [review]
Code looks good to me.
I have two more comments:
the 'g' in titles is still cut off a little, so I would suggest to increase the factor from 1.1 to 1.15 or 1.2
image
Also when you click on the HintButton of the Modules title while the details view is opened, the popover opens in the wrong place.
Screenshot 2024-06-13 at 18 49 36

File: src/shared/components/StatusBadge/StatusBadge.js [hunk 2]
```diff
@@ -14,29 +14,29 @@ const resolveType = status => {
     return undefined;
   }
 
-  switch (status.toUpperCase()) {
-    case 'INITIAL':
-    case 'PENDING':
-    case 'AVAILABLE':
-    case 'RELEASED':
+  switch (status) {
```
[the remark this answers is not in the document]
Comment (mrCherry97) [reply src/shared/components/StatusBadge/StatusBadge.js:17]: For extensions adjustment we need to ping the rest of the teams to adjust this.

**Review by akucharska — DISMISSED** [review]

File: public/i18n/en.yaml [hunk 3]
```diff
@@ -927,6 +927,8 @@ kubeconfig-id:
   error: "Couldn't load kubeconfig ID; configuration not changed (Error: ${{error}})"
   must-be-an-object: Kubeconfig must be a JSON or YAML object.
 kyma-modules:
+  module-added: Module added 
+  module-uninstall: Module uninstalled
```
Comment (grego952) [inline public/i18n/en.yaml:931]:
```suggestion
  module-uninstall: Module deleted
```

**Review by grego952 — APPROVED** [review]

**Review by OliwiaGowor — APPROVED** [review]

**Pull request merged**